src/utils/env.ts

```diff
@@ -11,9 +11,14 @@ const PC_ENV_TLDS = [
     'apne2.pure.cloud'
 ];
 
-let DEFAULT_PC_ENV = null;
-const PC_ENVS = PC_ENV_TLDS.reduce((result, currEnvTld) => {
-    let currEnv = {
+export interface PcEnv {
+    pcEnvTld: string;
+    pcAppOrigin: string;
+}
+
+let DEFAULT_PC_ENV!: PcEnv;
+const PC_ENVS = PC_ENV_TLDS.reduce<PcEnv[]>((result, currEnvTld) => {
```

Comment:
Sorry, a bit more ⛳ , a `.reduce` that just pushes onto a list looks like a `.map` to me.  Just with a bit of checking to set the `DEFAULT_PC_ENV` variable.

Maybe:

```ts
const PC_ENVS = PC_ENV_TLDS.map((currEnvTld) => ({
     pcEnvTld: currEnvTld,
     pcAppOrigin: `https://apps.${currEnvTld}`
});
// Might consider throwing an error rather than using `!` to assert that it exists.
const DEFAULT_PC_ENV = PC_ENVS.find(({pcEnvTld}) => pcEnvTld === DEFAULT_PC_ENV_TLD)!;
```

Reply:
Yeah, this one I ended up changing somewhere off in another PR and my change looks almost exactly like this haha